src/monitormanager.cpp

```diff
@@ -217,58 +217,42 @@ void MonitorManager::removeMonitor(HSMonitor* monitor)
 int MonitorManager::addMonitor(Input input, Output output)
 {
     // usage: add_monitor RECTANGLE [TAG [NAME]]
-    if (input.size() < 2) {
+    std::string cmd, rectString, tagName, monitorName;
+    input >> cmd >> rectString;
```

Comment:
While using the "old" interface, I actually liked that the command was (in a way) treated as a first-class citizen in ArgList, being available via the `command()` method, mainly for use in error messages. I didn't mind having to shift past the command before retrieving the first argument, but with the new stream operator, you even need a local variable to put it in.

So I'd suggest finishing what ArgList already started and turning it into an object that holds *a command and its arguments*, instead holding *a command and a list containing the command followed by its arguments*.

Reply:
I agree. So I propose to make Input a class that inherits from ArgList and has command name as a member. Actually, it is not clean that ArgList has a command() method.